agent/agent_test.go

```diff
@@ -649,8 +646,17 @@ func (s *AgentSuite) setupRunTask(projYml string) {
 	_, err := model.LoadProjectInto(s.ctx, []byte(projYml), nil, "", p)
 	s.Require().NoError(err)
 	s.tc.taskConfig.Project = p
-	s.tc.project = p
+
 	s.mockCommunicator.GetProjectResponse = p
+
+	t := &task.Task{
+		Id:           "task_id",
+		BuildVariant: "mock_build_variant",
+		DisplayName:  "this_is_a_task_name",
+		Version:      "my_version",
+	}
+	s.mockCommunicator.GetTaskResponse = t
+	s.tc.taskConfig.Task = t
```

Comment:
It seems like this is redundant because this task is a duplicate of the `s.task` already set in `SetupTest`.